src/hooks/useLocalStorageItem.ts

```diff
@@ -0,0 +1,43 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { tryParse } from "../libs/tryParse";
+
+
+// let myValue: [number, string];
+
+/**
+ * @param key Key to store in localStorage.
+ * @param blankValue Blank value to use before mounting.
+ * @param [initialValue] Value to fill in if not already stored.
+ * @returns Item and a function to update it.
+ * @remarks
+ * This does not attempt to retrieve from localStorage immediately,
+ * as doing so would cause a React SSR/client hydration mismatch.
+ */
+export function useLocalStorageItem<Value>(
+  key: string,
+  blankValue: Value,
+  initialValue = blankValue
+)  /* : [  Value,  (newValue: Value) => void,] */ {
+  const [item, setItem] = useState<Value>(blankValue);
+
+  useEffect(() => {
+    const retrievedValue = tryParse(localStorage.getItem(key)) as Value | undefined;
+    const actualNewValue = retrievedValue ?? initialValue;
+    setItem(actualNewValue);
+  }, [initialValue, key]);
+
+  const updateItem = useCallback(
+    (value: Value) => {
+      setItem(value);
+      // TODO: FIX THE VALUE STRING :( // @ts-expect-error
```

Comment:
In general, I would recommend _not_ changing runtime code when converting to TypeScript, as it's hard to keep track of both sets of changes. Consider using `// @ts-expect-error`s instead.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments